packages/app-cli/tests/models_Note.ts

```diff
@@ -125,6 +126,21 @@ describe('models_Note', function() {
 		expect(duplicatedNote.user_updated_time !== note1.user_updated_time).toBe(true);
 	}));
 
+	it('should duplicate a note with tags', (async () => {
+		const folder1 = await Folder.save({ title: 'folder1' });
+		const tag1 = await Tag.save({ title: 'tag1' });
+		const tag2 = await Tag.save({ title: 'tag2' });
+		const originalNote = await Note.save({ title: 'originalNote', parent_id: folder1.id });
+		await Tag.addNote(tag1.id, originalNote.id);
+		await Tag.addNote(tag2.id, originalNote.id);
+
+		const duplicatedNote = await Note.duplicate(originalNote.id);
+		const duplicatedNoteTags = await Tag.tagsByNoteId(duplicatedNote.id);
+
+		expect(duplicatedNoteTags.find(o => o.id === tag1.id)).toBeDefined();
```

Comment:
Thanks, and could you also please add a check on duplicatedNoteTags.length (which I guess should be 2)? Just to make sure it's not duplicating more than it should.

Reply:
A good idea